public/js/actions/types.js

```diff
@@ -113,13 +114,28 @@ type SourceAction =
                sourceText: SourceText }}
   | { type: "CLOSE_TAB", id: string };
 
+export type SidebarSide = "left" | "right";
+
+type SidebarCollapseAction = {
+  type: C.COLLAPSE_SIDEBAR,
+  side: SidebarSide,
+  collapsed: boolean
+};
+
+type SidebarResizeAction = {
+  type: C.RESIZE_SIDEBAR,
+  side: SidebarSide,
+  width: number
+};
```

Comment:
👍 I like the separate action types as we can type the reducer better in the future too